trimesh/util.py

```diff
@@ -1167,6 +1172,102 @@ def array_to_string(array,
 
     return formatted
 
+def structured_array_to_string(array,
+                    col_delim=' ',
+                    row_delim='\n',
+                    digits=8,
+                    value_format='{}'):
+    """
+    Convert an unstructured array into a string with a specified
+    number of digits and delimiter. The reason thisexists is
+    that the basic numpy array to string conversionsare
+    surprisingly bad.
+
+    Parameters
+    ------------
+    array : (n,) or (n, d) float or int
+       Data to be converted
+       If shape is (n,) only column delimiter will be used
+    col_delim : str
+      What string should separate values in a column
+    row_delim : str
+      What string should separate values in a row
+    digits : int
+      How many digits should floating point numbers include
+    value_format : str
+       Format string for each value or sequence of values
+       If multiple values per value_format it must divide
+       into array evenly.
+
+    Returns
+    ----------
+    formatted : str
+       String representation of original array
+    """
+    # convert inputs to correct types
+    array = np.asanyarray(array)
+    digits = int(digits)
+    row_delim = str(row_delim)
+    col_delim = str(col_delim)
+    value_format = str(value_format)
+
+    # abort for non-flat arrays
+    if len(array.shape) > 1:
```

Comment:
Note that it is assumed here that the second dimension is represented by the structured array elements